docs/install.md

```diff
@@ -424,6 +432,145 @@ Out-of-the-box, Tekton Pipelines Controller is configured for relatively small-s
+```
+
+which results in:
+
+```shell
+sha256:6fd8fe1d19933c62efe0195a05e2c93b5237739dc68c398843dfa2dd22fcb91d
+```
+
+Search the transparency log with the digest just collected:
+
+```shell
+rekor-cli search --sha sha256:6fd8fe1d19933c62efe0195a05e2c93b5237739dc68c398843dfa2dd22fcb91d
+```
+
+which results in:
+
+```shell
+Found matching entries (listed by UUID):
+eb69098b520a93ce140f4facfa596f4947ae7158910531d4882e466c24faec12
+257d788c3e8f3937e7085de5571ffd78d0156461eb7ca67baa9e9e0cdc7b4dc1
+```
+
+Tekton Chains generates provenance based the custom format in which the `subject` holds the list of artifacts which were
+built as part of the release. As part of the pipeline release, `subject` includes a list of images including pipeline
+controller image, pipeline webhook image, etc.
+
+```shell
+rekor-cli get --uuid eb69098b520a93ce140f4facfa596f4947ae7158910531d4882e466c24faec12 --format json | jq -r .Attestation | base64 --decode | jq
+```
+
+which results in:
+
+```shell
+{
+  "_type": "publish-release",
+  "predicateType": "https://tekton.dev/chains/provenance",
+  "subject": [
+    {
+      "name": "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/controller",
+      "digest": {
+        "sha256": "6fd8fe1d19933c62efe0195a05e2c93b5237739dc68c398843dfa2dd22fcb91d"
+      }
+    },
+    {
+      "name": "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/entrypoint",
+      "digest": {
+        "sha256": "61013ff4268fbf1046d623d3eab2bcb072e776202d86cb95ad7238a8c45c5420"
+      }
+    },
+    {
+      "name": "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init",
+      "digest": {
+        "sha256": "39b53980b255ef8af80086ac702f5b113d32f60cc4c8876b096ba37fc5912c81"
+      }
+    },
+    {
+      "name": "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/imagedigestexporter",
+      "digest": {
+        "sha256": "2aa8bc18e7d947c44b18be9b74241d508026c0d2b224577212cedb5a2347ad6f"
+      }
+    },
+    {
+      "name": "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/kubeconfigwriter",
+      "digest": {
+        "sha256": "12d08ecfb8bd5a635750909b2b1ea63454c97c77bef8aae7c0d6ea4308614c9d"
+      }
+    },
+    {
+      "name": "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/nop",
+      "digest": {
+        "sha256": "81c38a3d11e9f82898ac34318436b7019c23a2e279c60cb721a72c6ecccad476"
+      }
+    },
+    {
+      "name": "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/pullrequest-init",
+      "digest": {
+        "sha256": "57ec58d70867231a1d84e1b5f7298c6751859cd3426546c3ea7d759dffa8d1b2"
+      }
+    },
+    {
+      "name": "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/webhook",
+      "digest": {
+        "sha256": "f19dd16303ff8a8d55d706db5c8f8db593ba597684888bb15e70420fc4824103"
+      }
+    }
+  ],
+  ...
+```
+
+Now, you can verify each of these images using the Tekton public key:
```

Comment:
I don't fully understand the process.

- We start from the sha of the controller image taken from the release file.
- From there we search a matching provenance record. 
- We pick one of the two (does it matter which one? we do we get two?) 
- We look at the provenance record and get a list of the images in the release, with their shas
- We use `cosign` with the each of the images from the provenance record and match the output of cosign to the sha in the provenance record. This is done by adding the release tag to the image name

This way we check the images based on the tag, but we don't check the images in the `release.yaml` - apart perhaps from the controller one. Is that the intention? 

If Tekton users want to very a Tekton release before installing it from the `release.yaml` we should perhaps use the images from that file for the process?

Since the `release.yaml` itself is not signed we cannot guarantee that it has not been tampered with, but at least we can give users a way to check that the signature for all the images included in there is valid.

Reply:
> We start from the sha of the controller image taken from the release file.

Sha in the `release.yaml` is used to retrieve the provenance. Provenance collection can only be done using the digest (based on my understanding 😄 ).

> We pick one of the two (does it matter which one? we do we get two?)

Yup, we get two. Doesn't matter which one we choose, both results in the same record.

> We look at the provenance record and get a list of the images in the release, with their shas

> This way we check the images based on the tag, but we don't check the images in the release.yaml - apart perhaps from the controller one. Is that the intention?

That's a good point. I have added a step here to verify the deployment specifications in the `release.yaml` with the provenance collected before using the images.

Let me know how does it read now 🙏

Reply:
> We pick one of the two (does it matter which one? we do we get two?)

> Yup, we get two. Doesn't matter which one we choose, both results in the same record.

With the latest chains on dogfooding, the issue with multiple `UUID` is resolved. The provenance collection now returns a single `UUID`.  This is for the latest `v0.28.0`:

```shell
rekor-cli search --sha sha256:0c320bc09e91e22ce7f01e47c9f3cb3449749a5f72d5eaecb96e710d999c28e8
Found matching entries (listed by UUID):
68a53d0e75463d805dc9437dda5815171502475dd704459a5ce3078edba96226
```

Reply:
thanks @priyawadhwa, the issue with multiple `UUID` is resolved 🙏 

@afrittoli let me know what do you think about the step-by-step process now?